client/js/public.js

```diff
@@ -406,8 +406,42 @@
         return false;
       };
 
-      // keep a copy of the existing keys and remove updated ones
-      let toBeRemoved = new Set(scope.units.keys());
+      // while updating units and incidents, use a Set for the diff with existing ones
+      let toBeRemoved;
+      // incident data containes assigned units, also track the reverse direction
+      let incidents = new Map(); // unit.id => incident
```

Comment:
Although it is not used very often, it is possible that a unit is assigned to multiple incidents. Is this covered by this map implementation? I think we can get an inconsistent "blue", if one assigned incident is set "blue", and a second one of the same unit not.
I would expect it, that a unit is marked as "blueIncidentAssigned", if at least one of the assigned is blue.